### PR TITLE
Feature/authenticated user

### PR DESCRIPTION
**Pull request by JaimLus**
## Summary
- Spring security added

## Trello Card
https://trello.com/c/yOZWucp9/27-acceder-a-la-informaci%C3%B3n-del-user-logueado

### CODE REVIEW

File: src/main/java/wolox/training/controllers/UserController.java [hunk 1]
```diff
@@ -22,16 +23,31 @@
...
         this.bookRepository = bookRepository;
 
     }
 
+    /**
+     * Method that gets authenticated user
+     *
+     * @return username (String)
+     */
+    @GetMapping("/principal")
```
Comment (Jimenarosello) [inline src/main/java/wolox/training/controllers/UserController.java]: the most common naming strategies for the logged user info are "/me", "/user" 🙂 No need to change it, just a comment 😅

**Review by Jimenarosello — DISMISSED** [review]
Nice job! I left a comment regarding conventions, but you don't need to change it 😄

File: src/main/java/wolox/training/authentication/SpringContextAuthentication.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Component;
+
+@Component
+public class SpringContextAuthentication implements IAuthentication {
+
+    @Override
+    public Authentication getAuthentication() {
+
+        return SecurityContextHolder.getContext().getAuthentication();
```
Comment (marcegl91) [inline src/main/java/wolox/training/authentication/SpringContextAuthentication.java:13]: You can use the method shown in the section `3. Get the User in a Controller` of the documentation which is less code and there is no need to set up this class, but anyway this is ok.

**Review by marcegl91 — APPROVED** [review]
Great! :1st_place_medal:

**Review by Jimenarosello — APPROVED** [review]
🚀

**Pull request merged**